cpp/arcticdb/storage/s3/s3_storage.cpp

```diff
@@ -32,6 +34,48 @@ std::streamsize S3StreamBuffer::xsputn(const char_type* s, std::streamsize n) {
 }
 }
 
+void check_ec2_metadata_endpoint(const S3ApiInstance& s3_api) {
+    auto disabled_env = Aws::Environment::GetEnv("AWS_EC2_METADATA_DISABLED");
+    if (Aws::Utils::StringUtils::ToLower(disabled_env.c_str()) == "true") {
+        const char* who_disabled = s3_api.is_ec2_metadata_disabled_by_us() ?
+            "EC2 metadata endpoint did not respond in time so was bypassed":
+            "AWS_EC2_METADATA_DISABLED environment variable is set to true";
+        log::storage().warn("{}. This means machine identity authentication will not work.", who_disabled);
+    }
+}
+
+bool S3Storage::check_creds_and_bucket() {
+    using namespace Aws::S3;
+    // We cannot easily change the timeouts of the s3_client_, so use async:
+    auto future = s3_client_.HeadBucketCallable(Model::HeadBucketRequest().WithBucket(bucket_name_.c_str()));
```

Comment:
Do we need to respect `path_prefix` here? What if we have access to the bucket but don't have permissions for the root path.

Reply:
Recall that S3 doesn't actually have folders, so we can only reliably `HEAD` bucket or a full object name.

Reply:
That is true. If this is advisory I think it's OK.

Reply:
Is it true? I thought you absolutely could ACL specific directories in AWS S3, just not in the S3-like storages that we have (although it's on the roadmap for at least one of them)

Reply:
I think this is the best we can do. The prefix isn't a directory and will give another error if it's yet to be created presumably.